clicommand/agent_start.go

```diff
@@ -489,7 +489,11 @@ var AgentStartCommand = cli.Command{
 		defer done()
 
 		// Remove any config env from the environment to prevent them propagating to bootstrap
-		UnsetConfigFromEnvironment(c)
+		err = UnsetConfigFromEnvironment(c)
+		if err != nil {
+			fmt.Printf("%s", err)
+			os.Exit(1)
+		}
```

Comment:
exit feels a bit harsh, but we do it earlier in this method when a thing that should never happen ... happens. I followed the pattern.